### PR TITLE
[DDW-557] Implement block-consolidation status screen

### PR DESCRIPTION
**Pull request by daniloprates**
This PR implement a block-consolidation status screen:

The main idea is to create a screen which will show how many epochs have been already consolidated versus the total number of epochs.
The block-consolidation process is fully asynchronous and it processes/consolidates all the epochs except the 2 of the latest ones.
This process never ends but we can consider the case of having all the epochs consolidated except the 2 of the latest ones as 100% complete block consolidation.
In this simple screen we should also show:

* present the block-consolidation process progress (as a simple diagram or gauge)
* describe the process so that an average user can understand what it is all about
* provide helpful tips and links to the support portal articles in case user faces block-consolidation issues.

## Requirements

* The new screen should be accessible from the application menu under Block-consolidation status
* The new screen should be reachable regardless of the application status (e.g. if Daedalus is connected/synced/etc)

## Todo:

- [x] Create the screen
- [x] Add block consolidation logic
- [x] Add the menu item
- [x] Create [Storybook story](http://localhost:6006/?knob-Epochs%20Consolidated=57&knob-Epochs%20Synced=80&selectedKind=BlockConsolidationStatus&selectedStory=Default&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs)
- [x] English copy - @nikolaglumac 
- [x] Japanese translation - @nikolaglumac 
- [x] Add closing button (https://app.zeplin.io/project/57acfaa7436a750316161a8b/screen/5c3dd736dc2ae80e122db2ee?ref=slack)
- [x] Check why IPC communication for fetching the number of consolidated epochs is not working
- [x] Implement polling of current epoch and the number of consolidated epochs

## Screenshots:

image

comparison

---

### Basics

- [ ] PR is updated to the most recent version of target branch (and there are no conflicts)
- [ ] PR has good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance tests are passing (`yarn run test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn run dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn run package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn run flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn run lint`)
- [ ] Text changes are proofread and approved (Jane Wild)
- [ ] There are no missing translations (running `yarn run manage:translations` produces no changes)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn run storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly documented and commented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-rendering
- [ ] Any code that only works in Electron is neatly separated from components

### Testing
- [ ] New feature / change is covered by acceptance tests
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature / change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review:
- [ ] Merge PR
- [ ] Delete source branch
- [ ] Move ticket to `done` on the Youtrack board

### CODE REVIEW

**Comment by nikolaglumac** [comment]
@daniloprates I will send you the final text copy for this screen today.

**Comment by daniloprates** [comment]
progress

@a-rukin @nikolaglumac this is how I handled the "Epochs consolidated" label, so it doesn't stay on the left side, in the begining. Please, let me know if it's OK.

**Comment by daniloprates** [comment]
@a-rukin @nikolaglumac it's not immediately clear, from a user perspective, how to close this window. Maybe it could have a close button?

**Comment by daniloprates** [comment]
Still not clear the correct data for `devnet`. I've tried a few different `startTime` data:

* Looking at the [phyton file](https://iohk.myjetbrains.com/youtrack/_persistent/epoch.py?file=88-1457&c=false&updated=1547047001648) it uses a hardcoded start time for dev as well. Using this number will generate a `currentEpoch` number that is much larger than the other networks, but it seems to be the correct one.

* Using one of these will generate a `currentEpoch` number that is lower than the number of Epochs processed:
	* The manual `system start` time from `scripts/launch/demo-nix.sh`
	* The `_startTime` item from the `networkStatus` store
	* Using  or the  or 0 hour of present day 

Therefore, I'm using the first option, even if the `currentEpoch` is a huge number.

image

**Comment by daniloprates** [comment]
Still in `WIP` waiting for the Japanese translation, but ready to be reviewed.

**Comment by daniloprates** [comment]
@a-rukin 

> this text should be centered inside front square

What do you think of decreasing the opacity of the front square, a bit, for clarity?

**Comment by nikolaglumac** [comment]
@daniloprates you should try running Daedalus against staging network. In that case you don't have to run the demo cluster. You just need to start Daedalus with `NETWORK=staging yarn nix:staging`.

**Comment by nikolaglumac** [comment]
@daniloprates one last thing - before testing the Demo cluster you should delete the state directory so that you start clean! `rm -rf ~/Library/Application Support/Daedalus Demo/`

**Comment by daniloprates** [comment]
> @daniloprates the approach you took for the Demo/Development network is not good.
> The correct approach would be to use the `start time` from the `scripts/launch/demo-nix.sh`.
> In case this is hard (or not doable at all) we should use the start of the day (start of the day of the `_startTime` from `NetworkStatusStore` to be exact).
> In this case the target epoch (the network epoch) is 1 and you do not expect to consolidate that one.
> This is perhaps confusing but it is perfectly clear - the demo cluster always starts from zero blocks and generates 1 block every 20 seconds. One epoch lasts for 5 days! You would need to leave your demo cluster running for that long to see the epoch move to 2.

I've changed to use `_startTime` from `NetworkStatusStore` and will try to get `demo-nix.sh`'s tomorrow.

**Comment by nikolaglumac** [comment]
@daniloprates this will be released in Daedalus 0.13.0. We need to:
- merge latest develop into your branch
- change the target branch to `develop`

**Comment by daniloprates** [comment]
Please hold reviewing it for now.

**Comment by nikolaglumac** [comment]
@daniloprates we don't need to add acceptance tests for this one. At least not in the scope of this task. The only thing left to be done is to add some kind of a "Back" or "Close" button or some other way user can actually navigate away from the "Block consolidation" screen.
Perhaps @a-rukin can help us here?

close

File: source/main/utils/getNumberOfEpochsConsolidated.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import path from 'path';
+import { appFolderPath } from '../config';
+import { getNumberOfEpochsConsolidatedChannel } from '../ipc/getNumberOfEpochsConsolidated.ipc';
+import type { GetNumberOfEpochsConsolidatedChannelResponse } from '../../common/ipc/api';
+
+export const getNumberOfEpochsConsolidated = () => {
+  getNumberOfEpochsConsolidatedChannel
+    .onRequest((): Promise<GetNumberOfEpochsConsolidatedChannelResponse> => {
+      const epochsPath = path.join(appFolderPath, 'DB-1.0/epochs');
+      let epochsConsolidatedLength = 123;
```
[the remark this answers is not in the document]
Comment (nikolaglumac) [reply source/main/utils/getNumberOfEpochsConsolidated.js]: @daniloprates I have resolved this myself. Hope my fix is right :trollface:

File: source/renderer/app/stores/NetworkStatusStore.js [hunk 2]
```diff
@@ -112,6 +119,8 @@ export default class NetworkStatusStore extends Store {
     // Passively receive state changes of the cardano-node
     cardanoStateChangeChannel.onReceive(this._handleCardanoNodeStateChange);
 
+    this._getSystemStartTime();
```
Comment (nikolaglumac) [inline source/renderer/app/stores/NetworkStatusStore.js]: @daniloprates perhaps we can have this call only for development network? And please be careful - I'm not talking about `globals.environment.isDev` - that one compares NODE_ENV and we need `NETWORK === DEVELOPMENT` check. So please add the `const isDevelopment = NETWORK === DEVELOPMENT;` to `/source/main/environment.js` https://github.com/input-output-hk/daedalus/blob/develop/source/main/environment.js#L30

File: source/main/utils/getNumberOfEpochsConsolidated.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (daniloprates) [reply source/main/utils/getNumberOfEpochsConsolidated.js]: missed removing this test when IPC was driving me crazy. thanks

[the remark this answers is not in the document]
Comment (daniloprates) [reply source/main/utils/getNumberOfEpochsConsolidated.js]: > we should introduce pollers for both the number of consolidated epochs as well as for the current epoch value 

@nikolaglumac what interval you think?

[the remark this answers is not in the document]
Comment (nikolaglumac) [reply source/main/utils/getNumberOfEpochsConsolidated.js]: @daniloprates I think we should be fine with 20 sec intervals for both checks as that is the single slot duration.

File: source/renderer/app/stores/NetworkStatusStore.js (hunk 2 above)
Comment (daniloprates) [reply source/renderer/app/stores/NetworkStatusStore.js]: should I move `systemStartTime to `BlockConsolidationStatusPage` as well?

Comment (nikolaglumac) [reply source/renderer/app/stores/NetworkStatusStore.js]: No, I think it is fine to leave it in the NetworkStatusStore...

**Comment by daniloprates** [comment]
> Great work @daniloprates 🎉
> Everything works great - we just need to add the "back" button and then we are done.
> Thanks for adding this last piece of the puzzle 🙏

@nikolaglumac it's done. I also simplified the `TopBar` component, moving some logic to its container.

**Comment by nikolaglumac** [comment]
Good job @daniloprates!
Hope you made sure nothing got broken after the refactoring :))
There is just one thing I noticed - back arrow will always open the root route. Can we make it rather open the previous route?

**Comment by nikolaglumac** [comment]
@daniloprates @DominikGuzei @MarcusHurney let's wait with merging this one until @DmitriiGaico runs his tests on the installers 🙏

**Pull request merged**